README.md
# Programming and Problem Solving
#### By: Javier Orduz
[license-badge]: https://img.shields.io/badge/License-CC-orange
[license]: https://creativecommons.org/licenses/by-nc-sa/3.0/deed.en

[![CC License][license-badge]][license]  [![DS](https://img.shields.io/badge/downloads-DS-green)](https://github.com/Earlham-College/DS_Fall_2022)  [![Github](https://img.shields.io/badge/jaorduz-repos-blue)](https://github.com/jaorduz/)  ![Follow @jaorduc](https://img.shields.io/twitter/follow/jaorduc?label=follow&logo=twitter&logoColor=lkj&style=plastic)


This repo has material of my course on Programming 2023.

## Contents
1. [Python Basics](#pythonBasics)
1. [Folders](#folders)
1. [Exercises](#exercises)
1. [References](#references)

----------------

## 1. Python Basics <a name = pythonBasics></a>

Enclosed is supplementary material. I'll endeavor to compile all references; kindly flag any omissions of papers, repositories, or slides. I'm enthusiastic to incorporate your content into my documentation.

Extra material: 

1. [Harvard CS50’s Introduction to Programming with Python – Full University Course ](https://tinyurl.com/2pa4f3k5)

1. [Introduction to Data Science with Python](https://pll.harvard.edu/course/introduction-data-science-python)

<!-- []() -->

## 2. Folders <a name = folders></a>
### Students folder
Please use this folder to put your a clone/copy of your own repository.
[students repo](https://github.com/Earlham-College/CS128EC/tree/main/students)


# References <a name="references"></a>

[1] https://github.com/cs109/2015lab1
[2] https://harvard-iacs.github.io/2019-CS109A/pages/materials.html


history.txt
    0 This history was taken on Mond. Nov. 06. 2023
    1  pwd
    2  ls
    3  cd Physics/
    4  ls
    5  git clone https://github.com/jaorduc/PhysicsWEMO
    6  pwd
    7  git init
    8  git add .
    9  git commit -m "v0103102023"
   10  git add .
   11  git commit -m "v0103102023"
   12  git config --global user.email jaorduz@gmail.com
   13  git config --local user.email jaorduz@gmail.com
   14  git status
   15  git push origin master
   16  git config --global user.email jaorduz@gmail.com
   17  git config --local user.email jaorduz@gmail.com
   18  git config --global user.name "jaorduc"
   19  git config --local user.email "jaorduz@gmail.com"
   20  git init
   21  git add *
   22  git status
   23  git add
   24  git add .
   25  git commit -a
   26  git commit -a "v0203102022"
   27  git commit "v0203102022"
   28  git push origin
   29  git push --set-upstream origin master
   30  pwd
   31  ls
   32  pwd
   33  cd ..
   34  ls
   35  cd Physics/
   36  ls
   37  remove -r PhysicsWEMO/
   38  ls
   39  git remove rm PhysicsWEMO/
   40  git pull
   41  git fetche
   42  git fetch
   43  git push origin master
   44  pwd
   45  cd PhysicsWEMO/
   46  git push origin master
   47  git status
   48  git add EM_1.ipynb 
   49  git push origin master
   50  git add *
   51  git pull
   52  git push
   53  pwd
   54  cd ..
   55  ls
   56  pwd
   57  ls
   58  cd ..
   59  ls
   60  cd Physics/
   61  git push
   62  git remote add *
   63  git push
   64  git push jaorduz
   65  git push origin master
   66  git push
   67  pwd
   68  ls
   69  cd PhysicsWEMO/
   70  git oush
   71  git push
   72  pwd
   73  ls
   74  cd /srv/nbgrader/exchange/
   75  ls
   76  cd cs128-s23/
   77  ls
   78  cd ..
   79  sl
   80  ls
   81  pwd
   82  cd old/
   83  ls
   84  pwd
   85  cd
   86  ls
   87  git config --global user.name “Javier Orduz”
   88  git config --global user.email “jaorduz@ieee.org”
   89  git init 
   90  git clone https://github.com/Earlham-College/MATH120EC.git
   91  ls
   92  git push
   93  git push merge
   94  git push origin master
   95  git push --set-upstream merge master
   96  git init .
   97  git init .
   98  git push push origin master
   99  git push origin master
  100  git init .
  101  git push origin master
  102  pwd
  103  ls
  104  cd MATH120EC/
  105  ls
  106  git init .
  107  git push origin master
  108  git add .
  109  git push origin master
  110  ls
  111  git init 
  112  git add .
  113  cd MATH120EC/
  114  ls
  115  git init 
  116  git add .
  117  git remote add origin https://github.com/Earlham-College/MATH120EC.git
  118  git commit "v01"
  119  git init 
  120  git add .
  121  git pull origin master
  122  git log
  123  git pull origin master
  124  pwd
  125  ls
  126  git init -a
  127  git init
  128  git add .
  129  git pull 
  130  git push
  131  git init
  132  git add .
  133  git pull 
  134  git push
  135  git init
  136  git add .
  137  git pull 
  138  git push
  139  pwd
  140  ls
  141  cd MATH120EC/
  142  git init 
  143  git log
  144  git add .
  145  git commit -m "v01aug212023"
  146  git pull
  147  git push origin master
  148  git push -u origin master
  149  git status
  150  git push
  151  git merge
  152  git fetch
  153  git status
  154  git init 
  155  git add .
  156  git push
  157  git remote add origin master
  158  git remote add origin
  159  git remote add 
  160  git remote set-url
  161  git remote set-url --push 
  162  pwd
  163  history
  164  git clone https://github.com/Earlham-College/DS401EC.git
  165  pwd
  166  cd DS401EC/
  167  ls
  168  pwd
  169  git init 
  170  git add .
  171  git commit - m "v01augu222023"
  172  git commit -m "v01augu222023"
  173  git pull
  174  git push
  175  git push origin master
  176  git add .
  177  git commit -m "v01augu222023"
  178  git pull
  179  git push origin master
  180  git push
  181  cd ..
  182  ls
  183  cd MATH120EC/
  184  ls
  185  git init 
  186  git add .
  187  git commit -m "v01augu222023"
  188  git pull
  189  git push
  190  pwd
  191  ls
  192  cd ..
  193  ls
  194  nano readme.txt
  195  pwd
  196  ls
  197  git init
  198  git add .
  199  cd MATH120EC/
  200  git init 
  201  git add .
  202  git pull
  203  git push
  204  pwd
  205  cd ../DS401EC/
  206  git init 
  207  git add .
  208  git pull 
  209  git push 
  210  pwd
  211  ls
  212  cd DS401EC/
  213  git init 
  214  git add .
  215  git commit "v02augu222023"
  216  git commit -m "v02augu222023"
  217  git pull
  218  git push
  219  pwd
  220  cd ../MATH120EC/
  221  ls
  222  git init 
  223  git add .
  224  git commit "v02augu222023"
  225  git commit -m "v02augu222023"
  226  git pull
  227  git push
  228  git clone https://github.com/Earlham-College/CS128EC.git
  229  ls
  230  cp DS401EC/README.md CS128EC/
  231  cd CS128EC/
  232  ls
  233  pwd
  234  ls
  235  mkdir students
  236  ls
  237  cd students/
  238  mkdir group1
  239  mkdir group2
  240  mkdir group3
  241  ls
  242  mkdir group4
  243  mkdir group5
  244  mkdir group6
  245  mkdir group7
  246  mkdir group8
  247  mkdir group9
  248  mkdir group10
  249  ls
  250  pwd
  251  ls
  252  cd group1
  253  cd..
  254  ls
  255  pwd
  256  cd ..
  257  ls
  258  cd ..
  259  ls
  260  ¬pwd
  261  pwd
  262  ls
  263  git init
  264  git add .
  265  git commit -m "v01aug252023" 
  266  git pull
  267  git push
  268  cd students/group1
  269  nano readme.md
  270  cd ..
  271  cp group1/readme.md group2/
  272  cp group1/readme.md group3/
  273  cp group1/readme.md group4/
  274  cp group1/readme.md group5/
  275  cp group1/readme.md group6/
  276  cp group1/readme.md group7/
  277  cp group1/readme.md group8/
  278  cp group1/readme.md group9/
  279  cp group1/readme.md group10/
  280  pwd
  281  ls
  282  cd group5
  283  ls
  284  cd ..
  285  ls
  286  cd ..
  287  ls
  288  git init
  289  git add .
  290  git commit -m "v02aug252023"
  291  git pull
  292  git push

Tkinter/sampleC/windApp.py
#!/usr/bin/python
#-*- coding: latin-1 -*-
import wx

fileA = open("toWrite.txt","w")
class ExamplePanel(wx.Panel):
    def __init__(self, parent):
        wx.Panel.__init__(self, parent)
        self.quote = wx.StaticText(self, label="Fill the form", pos=(20, 30))
 
 
        self.button =wx.Button(self, label="Send", pos=(200, 200))
        self.Bind(wx.EVT_BUTTON, self.OnClick,self.button)
 
        self.lblname  = wx.StaticText(self, label="Name: ", pos=(20,60))
        self.editname = wx.TextCtrl(self, value="", pos=(150, 60), size=(250,-1))

        self.lblmail  = wx.StaticText(self, label="email: ", pos=(20,90))
        self.editmail = wx.TextCtrl(self, value="", pos=(150, 90), size=(250,-1))
 
        self.lblinsti = wx.StaticText(self, label="Institution:", pos=(20,120))
        self.editinsti = wx.TextCtrl(self, value="", pos=(150, 120), size=(250,-1))
#	    self.lbllevel  = wx.StaticText(self, label="Tu grado o carrera :", pos=(20,150))
        self.editlevel = wx.TextCtrl(self, value="", pos=(150, 150), size=(250,-1))


        self.quote = wx.StaticText(self, label="Check your email", pos=(20, 180))
    def OnClick(self,event):
        self.logger.AppendText("Click on the object %dn" % event.GetId())



app = wx.App(False)
# Create the father frame
frame = wx.Frame(None, title="Welcome Earlham College", size=(540,350))
# Create a note block
nb = wx.Notebook(frame)
# add the panels with Addpage
nb.AddPage(ExamplePanel(nb), "Data")

frame.Show()
app.MainLoop()